src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1555,12 +1555,50 @@ RebalanceTableShards(RebalanceOptions *options, Oid shardReplicationModeOid)
 		return;
 	}
 
+	/* find the name of the shard transfer mode to interpolate in the scheduled command */
+	Datum shardTranferModeLabelDatum =
+		DirectFunctionCall1(enum_out, shardReplicationModeOid);
+	char *shardTranferModeLabel = DatumGetCString(shardTranferModeLabelDatum);
+
+	if (list_length(placementUpdateList) == 0)
+	{
+		return;
+	}
+
+	/* schedule planned moves */
+	int64 jobId = CreateBackgroundJob("rebalance", "Rebalance colocation group ...");
+
+	PlacementUpdateEvent *move = NULL;
+	StringInfoData buf = { 0 };
+	initStringInfo(&buf);
+	bool first = true;
+	int64 prevJobId = 0;
+	foreach_ptr(move, placementUpdateList)
+	{
+		resetStringInfo(&buf);
+
+		appendStringInfo(&buf,
+						 "SELECT citus_move_shard_placement(%ld,%s,%u,%s,%u,%s)",
+						 move->shardId,
+						 quote_literal_cstr(move->sourceNode->workerName),
+						 move->sourceNode->workerPort,
+						 quote_literal_cstr(move->targetNode->workerName),
+						 move->targetNode->workerPort,
+						 quote_literal_cstr(shardTranferModeLabel));
+
+		BackgroundTask *task = ScheduleBackgroundTask(jobId, buf.data, first ? 0 : 1,
+													  &prevJobId);
+		prevJobId = task->taskid;
+		first = false;
+	}
+
 	/*
 	 * This uses the first relationId from the list, it's only used for display
 	 * purposes so it does not really matter which to show
 	 */
-	ExecutePlacementUpdates(placementUpdateList, shardReplicationModeOid, "Moving");
-	FinalizeCurrentProgressMonitor();
+
+	/* ExecutePlacementUpdates(placementUpdateList, shardReplicationModeOid, "Moving"); */
+	/* FinalizeCurrentProgressMonitor(); */
```

Comment:
Given the amount of new code we're introducing in risky areas (and the pg_cron BGW code is honestly not perfect, but we so far did not depend on it), it would be good preserve the existing rebalance_table_shards functionality and find a new (better!) name for the background rebalancer.

Reply:
I think this has been addressed in the code now (mostly - the new and old rebalanceer should still handle and prevent concurrent runs of one another)